integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java

```diff
@@ -148,7 +150,17 @@ public synchronized void startServerUsingCarbonHome(String carbonHome, Map<Strin
             waitTill(() -> !isRemotePortInUse("localhost", managementPort), 180, TimeUnit.SECONDS);
 
             if (!isRemotePortInUse("localhost", managementPort)) {
-                throw new RuntimeException("Server initialization failed");
+                if (retryAttempt < retryLimit){
+                    retryAttempt++;
+                    serverShutdown(portOffset, true);
+                    startServerUsingCarbonHome(carbonHome, commandMap);
+                }
+                else {
+                    throw new RuntimeException("Server initialization failed");
```

Comment:
do we need to set retryAttempt to zero here also ?

Reply:
done